src/osx/cocoa/dataview.mm

```diff
@@ -2910,16 +2912,45 @@ void UpdateWithRow(int row)
         }
     }
 
+    if ( attr.HasBackgroundColour() && [cell backgroundStyle] == NSBackgroundStyleLight )
```

Comment:
This is very minor, but I'd wrap `if ( [cell backgroundStyle] == NSBackgroundStyleLight )` check around both text and background colour-related code, it looks more logical like this.

Reply:
Done.